src/catchup/CatchupWork.cpp

```diff
@@ -351,6 +351,8 @@ CatchupWork::runCatchupStep()
 
         mApp.getHerder().setTrackingSCPState(catchupRange.last(),
                                              entry->header.scpValue);
+        // As we're catching up, do not claim to be in sync
+        mApp.getHerder().lostSync();
```

Comment:
Why do we need this? If catchup is triggered when "online" this is not necessary for sure.

Also, I am not sure we need to expose `lostSync` like this: I could see performing something like that in `setTrackingSCPState` if we force herder to "jump forward" for example.

Reply:
Note that this code doesn't run during online catchup due to the if condition - we only move Herder forward if it's behind. In case of online catchup, tracking should always be ahead or up to date (that's how we trigger catchup in the first place)

I agree that it's not the most intuitive way to manually call this function in catchup. I added this because I discovered a bug in my initial implementation. Specifically, when force-setting Herder to "tracking", I wasn't setting up out of sync timer, so the node would never go out of sync to read the latest ledgers from the network. So one option was to set Herder to tracking and set timers correctly. The other option was to start out of sync and let the timers be set when we hear from the network (this is similar to the startup code in Herder::restoreSCPState). I figured that for this use case it's unlikely that we'll be in sync after finishing catchup (the use case being captive core run with --start-at-ledger), but if you think otherwise, let me know.

Reply:
refactored this slightly: `setTrackingSCPState` can now set either tracking or syncing state, so external systems don't need to explicitly call lostSync. Note that this change made the startup path in `HerderImpl::start` a little cleaner too